### ISSUE TITLE
Terra Badge should have an aria-label attribute prop

### ISSUE THREAD

**Issue by JakeLaCombe** [cerner/terra-core#1823]
# Feature Request

Terra Badge has a visual indicator noting it's importance. However, screen readers will not know what to do with this element, and do nothing else besides read it as generic text. To improve screen reader support, an aria label prop should be added to the badge that identifies it's purpose.
